### PR TITLE
Google oauth

### PR DESCRIPTION
**Pull request by ayush893**

### CODE REVIEW

File: devsnest/app/controllers/api/v1/submissions_controller.rb [hunk 1]
```diff
@@ -13,9 +13,13 @@ def create
         choice = params['data']['attributes']['status']
 
         return render_error('User or Content not found') if user.nil? || content.nil?
+        
+        submission = Submission.create_submission(user.id, content.id, choice)
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/submissions_controller.rb:17]: it's already there, why have we repeated this line?

File: devsnest/app/controllers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	def google_oauth2
+	user = User.find_for_google_oauth2(request.env['omniauth.auth'],current_user)
+	if user.persisted?
+		render json: user
+	else
+		render "There was an error while trying to authenticate you..."
+	end
+
+	end
+end
```
Comment (vedansh) [inline devsnest/app/controllers/omniauth_callbacks_controller.rb:12]: Indent properly and leave a new line in the end

File: devsnest/app/views/public/homepage.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h1 class="mt-5"> Welcome to Yathartha Homeo Services </h1>
+
+<p class="lead"> please register or sign in to continue. </p>
+<%= link_to "sign Up", new_user_registration_path, class: "btn btn-success" %>
+<%= link_to "sign in", new_user_session_path, class: "btn btn-secondary" %>
```
Comment (vedansh) [inline devsnest/app/views/public/homepage.html.erb:5]: this is all temporary na?

File: devsnest/config/routes.rb [hunk 4]
```diff
@@ -10,16 +10,20 @@
...
-        collection do
-          get :report, :leaderboard
-        end
-      end
+  #     jsonapi_resources :users do
+  #       collection do
+  #         get :report, :leaderboard
+  #       end
+  #     end
+      #root to: "users#homepage"
```
Comment (vedansh) [inline devsnest/config/routes.rb:25]: why are we commenting this?

File: devsnest/db/seeds.rb [hunk 5]
```diff
@@ -64,3 +64,25 @@
 User.create(name: 'udit', discord_id: 'ebcd', password: '1234501', email: 'udit1251@gmail.com')
 User.create(name: 'vedansh', discord_id: 'ywer', password: '1237801', email: 'vedansh@gmail.com')
 
+=======
```
Comment (vedansh) [inline devsnest/db/seeds.rb:67]: Handle merge conflicts correctly, this is not good

File: devsnest/app/views/public/homepage.html.erb (hunk 3 above)
Comment (ayush893) [reply devsnest/app/views/public/homepage.html.erb:5]: yes, deleted

File: devsnest/Gemfile [hunk 6]
```diff
@@ -45,8 +45,11 @@ gem 'dotenv', '~> 2.7', '>= 2.7.6'
 gem 'rest-client', '~> 1.8'
 
 gem 'jsonapi-resources', '~> 0.10.2'
+gem 'devise'
```
Comment (vedansh) [inline devsnest/Gemfile:48]: always add a version number

File: devsnest/Gemfile [hunk 7]
```diff
@@ -27,7 +27,7 @@ gem 'bootsnap', '>= 1.4.2', require: false
 gem 'sidekiq', '~> 6.1', '>= 6.1.2'
 
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
-# gem 'rack-cors'
+gem 'rack-cors'
```
Comment (vedansh) [inline devsnest/Gemfile:30]: always add a version number

**Pull request closed without merging**